pkg/target/http.go

```diff
@@ -41,6 +44,11 @@ type HTTPTargetConfig struct {
 	CaFile                  string `hcl:"ca_file,optional" env:"TARGET_HTTP_TLS_CA_FILE"`
 	SkipVerifyTLS           bool   `hcl:"skip_verify_tls,optional" env:"TARGET_HTTP_TLS_SKIP_VERIFY_TLS"` // false
 	DynamicHeaders          bool   `hcl:"dynamic_headers,optional" env:"TARGET_HTTP_DYNAMIC_HEADERS"`
+
+	OAuth2ClientID     string `hcl:"oauth2_client_id,optional" env:"TARGET_HTTP_OAUTH2_CLIENT_ID"`
```

Comment:
If you make these *string, you can have them be nil instead of empty strings for the check below.

Doesn't make any real difference to the code, I think we don't use one or the other consistently across the app - but in case you didn't know that pattern. :)

Reply:
I think in most places (at least in http target!) we use empty strings so I'm going to leave it as it now :)